segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/bookkeeper/Ledgers.java

```diff
@@ -15,43 +15,65 @@
-import org.apache.bookkeeper.client.BookKeeper;
-import org.apache.bookkeeper.client.LedgerHandle;
+import org.apache.bookkeeper.client.BKException.BKNotEnoughBookiesException;
+import org.apache.bookkeeper.client.BKException.BKUnexpectedConditionException;
+import org.apache.bookkeeper.client.api.BookKeeper;
+import org.apache.bookkeeper.client.api.DigestType;
+import org.apache.bookkeeper.client.api.Handle;
+import org.apache.bookkeeper.client.api.ReadHandle;
+import org.apache.bookkeeper.client.api.WriteHandle;
+import org.apache.bookkeeper.common.concurrent.FutureUtils;
 
 /**
  * General utilities pertaining to BookKeeper Ledgers.
  */
 @Slf4j
 final class Ledgers {
-    static final long NO_ENTRY_ID = LedgerHandle.INVALID_ENTRY_ID;
+    static final long NO_ENTRY_ID = -1;
     /**
      * How many ledgers to fence out (from the end of the list) when acquiring lock.
      */
     static final int MIN_FENCE_LEDGER_COUNT = 2;
-    private static final BookKeeper.DigestType LEDGER_DIGEST_TYPE = BookKeeper.DigestType.MAC;
+    private static final DigestType LEDGER_DIGEST_TYPE = DigestType.MAC;
+
+    /**
+     * Converts exceptions to BKException. Fallback to BKUnexpectedConditionException
+     * for non BookKeeper exceptions.
+     */
+    private static final Function<Throwable, BKException> BK_EXCEPTION_HANDLER = cause -> {
+        if (cause instanceof BKException) {
+            return (BKException) cause;
+        } else {
+            BKUnexpectedConditionException wrapper = new BKUnexpectedConditionException();
+            wrapper.initCause(cause);
+            return wrapper;
+        }
+    };
 
     /**
      * Creates a new Ledger in BookKeeper.
      *
-     * @return A LedgerHandle for the new ledger.
+     * @return A WriteHandle for the new ledger.
      * @throws DataLogNotAvailableException If BookKeeper is unavailable or the ledger could not be created because an
      *                                      insufficient number of Bookies are available. The causing exception is wrapped
      *                                      inside it.
      * @throws DurableDataLogException      If another exception occurred. The causing exception is wrapped inside it.
      */
-    static LedgerHandle create(BookKeeper bookKeeper, BookKeeperConfig config) throws DurableDataLogException {
+    static WriteHandle create(BookKeeper bookKeeper, BookKeeperConfig config) throws DurableDataLogException {
         try {
             return Exceptions.handleInterruptedCall(() ->
-                    bookKeeper.createLedger(
-                            config.getBkEnsembleSize(),
-                            config.getBkWriteQuorumSize(),
-                            config.getBkAckQuorumSize(),
-                            LEDGER_DIGEST_TYPE,
-                            config.getBKPassword()));
-        } catch (BKException.BKNotEnoughBookiesException bkEx) {
+                    FutureUtils.result(bookKeeper.newCreateLedgerOp()
```

Comment:
please note that FutureUtils#result already handles InterruptedException in the same way Exception#handleInterruptedCall does

https://github.com/apache/bookkeeper/blob/master/bookkeeper-common/src/main/java/org/apache/bookkeeper/common/concurrent/FutureUtils.java#L73

I prefer leaving Pravega Exceptions#handleIntettuptedCall in place in order to make it clear that we are dealing with InterruptedException, but actually it is not needed